url29ecd5d728_geodb_db3dcore_tgz-pJ8-de_uos_igf_db3d_dbms_model3d_SegmentNetBuilderJ8/src/de/uos/igf/db3d/dbms/newModel4d/ServicesFor4DObjects.java

```diff
@@ -0,0 +1,294 @@
+package de.uos.igf.db3d.dbms.newModel4d;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import de.uos.igf.db3d.dbms.geom.Point3D;
+import de.uos.igf.db3d.dbms.geom.ScalarOperator;
+import de.uos.igf.db3d.dbms.model3d.ComplexGeoObj;
+import de.uos.igf.db3d.dbms.model3d.Object3D;
+import de.uos.igf.db3d.dbms.model3d.Object3DBuilder;
+import de.uos.igf.db3d.dbms.model3d.PointElt3D;
+import de.uos.igf.db3d.dbms.model3d.PointNetBuilder;
+import de.uos.igf.db3d.dbms.model3d.SegmentElt3D;
+import de.uos.igf.db3d.dbms.model3d.SegmentNetBuilder;
+import de.uos.igf.db3d.dbms.model3d.TetrahedronElt3D;
+import de.uos.igf.db3d.dbms.model3d.TetrahedronNetBuilder;
+import de.uos.igf.db3d.dbms.model3d.TriangleElt3D;
+import de.uos.igf.db3d.dbms.model3d.TriangleNetBuilder;
+import de.uos.igf.db3d.dbms.structure.Space3D;
+
+public class ServicesFor4DObjects {
+
+	/**
+	 * The 4D-to-3D Service
+	 * 
+	 * The main service for 4D objects which creates suitable 3D objects at the
+	 * specified time
+	 * 
+	 * Note: A Object4D can only have one type of Spatial Part!
+	 * 
+	 * @param date
+	 *            - the effective date for the object3D which will be created
+	 * @return the Object3D object which represents the Object4D at the
+	 *         specified date
+	 * @author Paul Vincent Kuper (kuper@kit.edu)
+	 */
+	public static Object3D getInstanceAt(Object4D object, Date date) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.